packages/expo-modules-autolinking/scripts/ios/react_import_patcher.rb

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018-present 650 Industries. All rights reserved.
+
+module Expo
+  class ReactImportPatcher
+
+    public def initialize(installer, options)
+      @root = installer.sandbox.root
+      @module_dirs = get_module_dirs(installer)
+      @options = options
+    end
+
+    public def run!
+      args = [
+        'node',
+        '--eval',
+        'require(\'expo-modules-autolinking\')(process.argv.slice(1))',
+        'patch-react-imports',
+        '--pods-root',
+        Shellwords.escape(File.expand_path(@root)),
+      ]
+
+      if @options[:dry_run]
+        args.append('--dry-run')
+      end
+
+      @module_dirs.each do |dir|
+        args.append(Shellwords.escape(File.expand_path(dir)))
+      end
+      Pod::UI.message "Executing ReactImportsPatcher node command: #{args.join(' ')}"
+
+      time_begin = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+      system(*args)
+      elapsed_time = Process.clock_gettime(Process::CLOCK_MONOTONIC) - time_begin
+      Pod::UI.info "ReactImportsPatcher takes #{elapsed_time.round(4)} seconds to transform files."
+    end
+
+    private def get_module_dirs(installer)
+      unless installer.pods_project
+        Pod::UI.message '`pods_project` not found. This happens possibly from `incremental_installation`'
```

Comment:
could you explain more about how this error could happen?

Reply:
take the bare-expo as an example which has the `incremental_installation` in Podfile. `pod install` at the second time, CocoaPods will not generate Pods project again and the `installer.pods_project` will be nil here. i think this case is fine because we will do the patch at the first time of `pod install`.

`Pod::UI.message` will show only if user has `pod install --verbose`. that's kinda a debug log here. if you think we'd better to have more clear in the log message, please let me know.

Reply:
Yeah, this makes sense and will only be printed in verbose mode. Logging system in CocoaPods is kinda weird — the actual log level and verbosity is determined based on how deep your message call is. As far as I remember it's wrapped by another `Pod::UI.message 'parent message' do ... end`, so its log level is verbose.